### PR TITLE
Implemented Packable for Vec

### PR DESCRIPTION
**Pull request by zesterer**
# Description of change

Implemented `Packable` for `Vec<T: Packable>`.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Tests are included in the PR.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Review by Alex6323 — DISMISSED** [review]

File: bee-common/bee-common/tests/packable.rs [hunk 1]
```diff
@@ -53,3 +53,18 @@ fn packable_option() {
         Some(42u64)
     );
 }
+
+#[test]
+fn package_vector() {
```
Comment (thibault-martinez) [inline bee-common/bee-common/tests/packable.rs]:
```suggestion
fn packable_vector() {
```

Comment (zesterer) [reply bee-common/bee-common/tests/packable.rs]: Whoops, will change

Comment (Alex6323) [reply bee-common/bee-common/tests/packable.rs]: eagle eyes :smile:

File: bee-common/bee-common/src/packable.rs [hunk 2]
```diff
@@ -50,6 +50,29 @@ impl Packable for bool {
     }
 }
 
+impl<T: Packable> Packable for Vec<T>
```
Comment (thibault-martinez) [inline bee-common/bee-common/src/packable.rs]: Would you mind changing the type parameter to `P` like the Option implementation ?

**Review by samuel-rufi — DISMISSED** [review]

**Review by thibault-martinez — DISMISSED** [review]

**Review by Alex6323 — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**